yarn-project/p2p/src/services/peer-manager/peer_manager.ts

```diff
@@ -168,8 +184,13 @@ export class PeerManager implements PeerManagerInterface {
     } else {
       this.logger.verbose(`Connected to transaction peer ${peerId.toString()}`);
     }
-    if (!this.config.p2pDisableStatusHandshake) {
+    if (this.config.p2pDisableStatusHandshake) {
+      return;
+    }
```

Comment:
I think we can remove this. It was added by @spalladino because of flaky tests. However, the flakes were caused by incorrect stream ID handling.

Reply:
I'll leave that for another PR